tests/test_utils.py

```diff
@@ -32,3 +32,11 @@ def test_check_embeddings_shape():
     embeddings = np.array([[1, 2, 3],
                            [2, 3, 4]])
     check_embeddings_shape(embeddings, docs)
+
+
+def test_make_unique_distances():
+    dists = np.array([0, 0, 0.5, 0.75, 1, 1])
+    unique_dists = get_unique_distances(dists)
+
+    assert len(unique_dists) == len(dists), "The number of elements must be the same"
+    assert len(dists) == len(np.unique(unique_dists)), "The distances must be unique"
```

Comment:
As mentioned above, the noise might need to be smaller to keep the current order as this example shows large differences in distance between subsequent values. As a result, this test might be too lenient.

Reply:
That was an issue in the code as mentioned in the comment below.